src/main/webapp/search_area.js

```diff
@@ -12,81 +12,96 @@
     this.form = document.createElement("form");
-    this.form.setAttribute("action", "/list-organizations");
-    this.form.setAttribute("method", "POST");
+    this.form.setAttribute("onsubmit", "return false");
 
     this.zipcodeFormLabel = document.createElement("label");
     this.zipcodeFormLabel.setAttribute("for", "zipcode-entry");
     this.zipcodeFormLabel.setAttribute("id", "zipcode-label");
     this.zipcodeFormLabel.textContent = "Enter a zipcode: ";
-    this.zipcodeForm.appendChild(this.zipcodeFormLabel);
+    this.form.appendChild(this.zipcodeFormLabel);
 
     this.zipcodeFormEntry = document.createElement("input");
     this.zipcodeFormEntry.setAttribute("type", "text");
     this.zipcodeFormEntry.setAttribute("id", "zipcode-entry");
     this.zipcodeFormEntry.setAttribute("pattern", "[0-9]{5}");
     this.zipcodeFormEntry.setAttribute("name", "zipcode");
-    this.zipcodeForm.appendChild(this.zipcodeFormEntry);
+    this.form.appendChild(this.zipcodeFormEntry);
 
     this.zipcodeSubmit = document.createElement("input");
     this.zipcodeSubmit.setAttribute("type", "submit");
     this.zipcodeSubmit.setAttribute("class", "gray-button");
-    this.zipcodeForm.appendChild(this.zipcodeSubmit);
-
-    this.organizationSearchArea.appendChild(this.zipcodeForm);
-
-    this.filterButtonArea = document.createElement("div");
-    this.filterButtonArea.setAttribute("id", "filter-button-area");
-
-    this.filterButton = document.createElement("button");
-    this.filterButton.setAttribute("type", "button");
-    this.filterButton.setAttribute("class", "gray-button");
-    this.filterButton.setAttribute("id", "filter-button");
-    this.filterButton.textContent = "Filter";
-    this.filterButtonArea.appendChild(this.filterButton);
-
-    this.organizationSearchArea.appendChild(this.filterButtonArea);
-
-    this.organizationList = document.createElement("div");
-    this.organizationList.setAttribute("id", "organization-list");
+    this.zipcodeSubmit.addEventListener('click', () => this.setUrlParamValue("zipcode", this.form.zipcode.value));
+    this.form.appendChild(this.zipcodeSubmit);
+
+    this.zipcodeFormArea.appendChild(this.form);
+    this.organizationSearchArea.appendChild(this.zipcodeFormArea);
+
+    this.filterInputArea = document.createElement("input");
+    this.filterInputArea.setAttribute("list", "filter-datalist");
+    this.filterInputArea.setAttribute("id", "filter-input-area");
+    this.filterInputArea.setAttribute("placeholder", "Filter Results");
+    this.filterInputArea.addEventListener('keypress', function (e) {
+      if (e.key === 'Enter') {
+        this.setUrlParamValue("filterParam", this.filterInputArea.value);
+      }
+    }.bind(this));
+
+    this.filterDataList = document.createElement("datalist");
+    this.filterDataList.setAttribute("id", "filter-datalist");
+    this.filterOptions = ["Foods", "Clothing", "Shelter"];
+    for (let i = 0; i < this.filterOptions.length; i++) {
```

Comment:
If you only refer to values but never the index you can write

for (const value of this.filterOptions) {
  const option = ..;
  option.value = value;
  ...
}